api/src/main/java/com/velocitypowered/api/proxy/server/PingOptions.java

```diff
@@ -0,0 +1,157 @@
+ * reference the LICENSE file in the api top-level directory.
+ */
+
+package com.velocitypowered.api.proxy.server;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import com.velocitypowered.api.network.ProtocolVersion;
+import java.time.Duration;
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+import net.kyori.adventure.builder.AbstractBuilder;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Contains the parameters used to ping a {@link RegisteredServer}.
+ * This class is immutable.
+ *
+ * @since 3.2.0
+ * @see RegisteredServer#ping(PingOptions)
+ */
+public final class PingOptions {
+  /**
+   * Default PingOptions.
+   * Uses a {@link ProtocolVersion#UNKNOWN} version and a timeout of 3000.
+   */
+  public static final PingOptions DEFAULT = PingOptions.builder().build();
+  private final ProtocolVersion protocolVersion;
+  private final long timeout;
+
+  private PingOptions(final Builder builder) {
+    this.protocolVersion = builder.protocolVersion;
+    this.timeout = builder.timeout;
+  }
+
+  /**
+   * The protocol version used to ping the server.
+   *
+   * @return the emulated Minecraft version
+   */
+  public ProtocolVersion getProtocolVersion() {
+    return this.protocolVersion;
+  }
+
+  /**
+   * The maximum period of time to wait for a response from the remote server.
+   *
+   * @return the server ping timeout in milliseconds
+   */
+  public long getTimeout() {
+    return this.timeout;
+  }
+
+  /**
+   * Create a new builder to assign values to a new PingOptions.
+   *
+   * @return a new {@link PingOptions.Builder}
+   */
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (o == null) {
+      return false;
+    }
+    if (!(o instanceof PingOptions)) {
+      return false;
+    }
+    final PingOptions other = (PingOptions) o;
+    return Objects.equals(this.protocolVersion, other.protocolVersion)
+            && Objects.equals(this.timeout, other.timeout);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(this.protocolVersion, this.timeout);
+  }
+
+  @Override
+  public String toString() {
+    return "PingOptions{"
+            + "protocolVersion=" + protocolVersion
+            + ", timeout=" + timeout
+            + '}';
+  }
+
+  /**
+   * A builder for {@link PingOptions} objects.
+   *
+   * @since 3.2.0
+   */
+  public static final class Builder implements AbstractBuilder<PingOptions> {
+    private ProtocolVersion protocolVersion;
+    private long timeout = 3000;
+
+    private Builder() {
+    }
+
+    /**
+     * Sets the protocol with which the server is to be pinged.
+     *
+     * @param protocolVersion the specified protocol
+     * @return this builder
+     */
+    public Builder version(final ProtocolVersion protocolVersion) {
+      this.protocolVersion = protocolVersion;
```

Comment:
Check for null here and initialize the version class field to `UNKOWN`; that way we don't need the null check in the `build` method.